Servers/UI/OJS.Servers.Ui/ClientApp/src/components/submissions/details/SubmissionDetails.tsx

```diff
@@ -41,12 +47,18 @@ const SubmissionDetails = () => {
             return;
         }
 
-        const { problem: { id: problemId }, isOfficial } = currentSubmission;
+        const { problem: { id: problemId }, isOfficial, user: { id: userId } } = currentSubmission;
 
-        (async () => {
-            await getSubmissionResults(problemId, isOfficial);
-        })();
-    }, [ currentSubmission, getSubmissionResults ]);
+        if (!user.permissions.canAccessAdministration) {
+            (async () => {
+                await getSubmissionResults(problemId, isOfficial);
+            })();
+        } else {
+            (async () => {
+                await getSubmissionResultsByProblemAndUser(problemId, isOfficial, userId);
+            })();
+        }
```

Comment:
This is business logic. Should be moved inside the hook.

Reply:
Done.